Sprint_6

chunkychunk2 opened the pull request:
complete homework and add jacoco report

chunkychunk2 committed:
complete homework and add jacoco report

mkouznetsov-git commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,36 @@
+    @Test
+    public void getSoundTest() throws Exception {
+        Cat cat = new Cat(feline);
+       // Mockito.verify(cat, Mockito.times(1)).getSound();
```
Ненужные комментарии следует удалять из финальной версии кода.

mkouznetsov-git commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,36 @@
+public class CatTest {
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
```
Если в этом классе не используется мокирование, то это здесь не нужно

mkouznetsov-git commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,36 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
```
Много неиспользуемых импортов -- удаляй.

mkouznetsov-git commented on src/test/java/com/example/FelineParamTest.java:
```diff
@@ -0,0 +1,59 @@
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.List;
```
Импорт не используется

mkouznetsov-git commented on src/test/java/com/example/FelineParamTest.java:
```diff
@@ -0,0 +1,59 @@
+    Feline feline;
+
+    @Mock
+    Animal animal;
```
Переменная инициализируется, но не используется

mkouznetsov-git commented on src/test/java/com/example/FelineTest.java:
```diff
@@ -0,0 +1,49 @@
+
+
+    @Mock
+    Feline feline;
```
Нет смысла мокировать тестируемый объект

mkouznetsov-git commented on src/test/java/com/example/LionParamTest.java:
```diff
@@ -0,0 +1,56 @@
+    Feline feline;
+
+    @Test
+    public void getFoodMockTest() throws Exception {
```
Если наша тестируемая программа будет глючить и будет кидать exception даже на корректных значениях пола для льва, то такой тест будет проходить успешно и пропускать эти ошибки.

mkouznetsov-git commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,31 @@
+public class LionTest {
+
+    @Mock
+    Lion lion;
```
Не нужно мокировать тестируемый объект

chunkychunk2 committed:
Исправил ошибки

mkouznetsov-git commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    @Test
+    public void doesHaveManeMockTest() throws Exception {
```
Успешные тесты на гриву находятся в другом классе, зачем здесь дублирование? Здесь, наверное, должна быть проверка getFood?

mkouznetsov-git commented on src/test/java/com/example/LionParamTest.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    @Mock
+    Feline feline;
```
Повторно: это некорректное поведение проверки. Если класс льва будет глючить и выдавать исключение на ЛЮБОЕ значение пола, то тест всё-равно будет проходить успешно.

mkouznetsov-git commented on src/test/java/com/example/LionParamTest.java:
```diff
@@ -0,0 +1,61 @@
+    public static Object[][] getData() {
+        return new Object[][]{
+                {"Самец", true},
+                {"самей", false},
```
Некорректное значение пола не приводит ни к true, ни в false. Оно приводит к исключению. Поэтому этот кейс неверно проверять таким образом. Его нужно проверять отдельно от параметризованного класса -- отдельным тестовым методом. Перенеси эту проверку в класс LionTest.

chunkychunk2 committed:
Перенес тест на пол в LionTest
Убрал тесты на гриву, замеменил на проверку getFood

mkouznetsov-git commented on src/test/java/com/example/LionParamTest.java:
```diff
@@ -0,0 +1,32 @@
+
+import java.util.List;
+
+public class LionParamTest {
```
Ну теперь это вообще не параметризованный тест. По идее здесь должны быть 2 параметра пола для проверки гривы.

mkouznetsov-git commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Test
+    public void getSexMockTest() throws Exception {
```
Это тест из списка параметризованных, зачем ты его сюда перенёс. Он должен быть в классе LionParamTest. Только проверяться должен с использованием параметризации.

mkouznetsov-git commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,38 @@
+        Lion lion = new Lion("Самка", feline);
+        Assert.assertEquals(List.of("Животные", "Птицы", "Рыба"), lion.getFood());
+    }
+
```
Где-то по дороге потерялся тест на невалидное значение пола.

mkouznetsov-git commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Test
+    public void doesHaveFoodMockTest() throws Exception {
```
Ты пишешь в названии теста MockTest, но мокирование не используешь. Там, где ты не используешь мокирование, не нужно называть методы Mock. Либо задействуй мокирование.

mkouznetsov-git commented on src/test/java/com/example/LionParamTest.java:
```diff
@@ -0,0 +1,32 @@
+    public void getFoodMockTest() throws Exception {
+            Lion lion = new Lion("Самец", feline);
+            Mockito.when(feline.getFood("Хищник")).thenReturn(List.of("Животные", "Птицы", "Рыба"));
+            Assert.assertEquals(List.of("Животные", "Птицы", "Рыба"), lion.getFood());
```
Если это тест не использует параметризованные значения, то он не должен находиться в классе с параметризацией.

chunkychunk2 committed:
Параметризировал тесты на гриву.
Переменовал тесты с ключевым словом мок, где они используются, а где не используются убрал.
Вынес в LionTest проверку с исключением пола.
Убрал импорт неиспользуемых библиотек.

mkouznetsov-git commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,25 @@
+    Feline feline;
+
+    @Test
+    public void getSoundMockTest() throws Exception {
```
Опять слово mock в названии теста, но мокирование не используется. Тебе надо разобраться с этой теорией и переложить на практику. Туда-сюда менять местами названия и аннотации нет смысла.

mkouznetsov-git commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,25 @@
+    @Test
+    public void getFoodMockTest() throws Exception {
+        Cat cat = new Cat(feline);
+        Assert.assertTrue(cat.getFood().isEmpty());
```
Почему мы ожидаем, что getFood() для кота должен быть пустым?

mkouznetsov-git commented on src/test/java/com/example/LionParamTest.java:
```diff
@@ -0,0 +1,50 @@
+    @Test
+    public void haveMineMockTest() throws Exception {
+        Lion lion = new Lion(sex, feline);
+        Mockito.when(mockLion.doesHaveMane()).thenReturn(mane);
```
Третий раз повторю: нет смысла мокировать тестируемый объект. Что ты проверяешь таким образом?

chunkychunk2 committed:
Убрал моки, где нет необходимости.
Переписал метод getFood() для кота.
Изменил названия методов с ключевым словом Mock, где они используются, а где нет - убрал.

mkouznetsov-git commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,29 @@
+@RunWith(MockitoJUnitRunner.class)
+public class CatTest {
+    @Mock
+    Cat cat;
```
Четвёртый раз :)
Не нужно мокировать тестируемый объект!

chunkychunk2 committed:
Убрал мок в тестируемом объекте

chunkychunk2 committed:
Убрал мок в тестируемом объекте
